sdks/typescript/src/v1/declaration.ts

```diff
@@ -439,3 +557,41 @@ export function CreateWorkflow<T extends JsonObject = any, K extends JsonObject
 ): WorkflowDeclaration<T, K> {
   return new WorkflowDeclaration<T, K>(options, client);
 }
+
+/**
+ * Creates a new durable task workflow declaration with types inferred from the function parameter.
+ * @template Fn The type of the durable task function
+ * @param options The durable task configuration options.
+ * @param client Optional Hatchet client instance.
+ * @returns A new TaskWorkflowDeclaration with inferred types.
+ */
+export function CreateDurableTaskWorkflow<
+  // Extract input and return types from the function, but ensure they extend JsonObject
+  Fn extends (input: I, ctx: DurableContext<I>) => O | Promise<O>,
+  I extends JsonObject = Parameters<Fn>[0],
+  O extends JsonObject = ReturnType<Fn> extends Promise<infer P>
+    ? P extends JsonObject
+      ? P
+      : never
+    : ReturnType<Fn> extends JsonObject
+      ? ReturnType<Fn>
+      : never,
+>(
+  options: {
+    fn: Fn;
+  } & Omit<CreateWorkflowDurableTaskOpts<I, O>, 'fn'>,
+  client?: IHatchetClient
+): TaskWorkflowDeclaration<I, O> {
+  // Note: We're using TaskWorkflowDeclaration here since task and durableTask
+  // share the same declaration structure but with different task types
+  const taskWorkflow = new TaskWorkflowDeclaration<I, O>(options as any, client);
+
+  // Move the task from tasks to durableTasks
+  if (taskWorkflow.definition._tasks.length > 0) {
+    const task = taskWorkflow.definition._tasks[0];
+    taskWorkflow.definition._tasks = [];
+    taskWorkflow.definition._durableTasks.push(task as any);
```

Comment:
I'm not totally clear on why we need this cast to any (also a bunch of similar type casts in other places) - is `_tasks` not an array of `Task` and compatible with `_durableTasks`?

Reply:
good catch!